src/cellEditors/CellEditor.js

```diff
@@ -224,16 +229,54 @@ var CellEditor = Base.extend('CellEditor', {
     },
 
     /**
+     * Saves the edited value by calling the {@link CellEditor#saveEditorValue|saveEditorValue} method.
+     *
+     * Before saving, performs a validity check which consists of:
+     * 1. Call `editorValueIsValid`, which calls the localizer's `isValid()` function, if available.
+     * 2. Catch any errors thrown by the {@link CellEditor#getEditorValue|getEditorValue} method.
+     * 3. Check return value for `NaN`.
+     *
+     * If the validity check fails, the effect indicated in the {@link CellEditor#errorEffect|errorEffect} property is called. After three such calls in a cell editor function, an instructional alert is displayed.
+     * @returns {boolean} Truthy means successful stop. Falsy means syntax error prevented stop.
      * @memberOf CellEditor.prototype
      * @desc stop editing
      */
     stopEditing: function() {
-        var value = this.getEditorValue();
-        if (this.grid.fireSyntheticEditorDataChangeEvent(this, this.initialValue, value)) {
+        var valid = this.editorValueIsValid();
```

Comment:
What kind of configuration did you have in mind. As I see it, all this functionality applies to all cell editors. That said, {{CellEditor}} is an "abstract base class" (quotes because this is JS so not really) and {{stopEditing}} (like any other method) is overrideable in the implementing class.